sdk/core/azure-core/src/main/java/com/azure/core/http/rest/RestProxy.java

```diff
@@ -346,7 +347,8 @@ private static Exception instantiateUnexpectedException(UnexpectedExceptionInfor
         if ("application/octet-stream".equalsIgnoreCase(contentType)) {
             bodyRepresentation = "(" + httpResponse.getHeaderValue("Content-Length") + "-byte body)";
         } else {
-            bodyRepresentation = responseContent.isEmpty() ? "(empty body)" : "\"" + responseContent + "\"";
+            responseContent = ImplUtil.printPrettyFormatJsonOrXml(responseContent, contentType);
```

Comment:
You should only pay the performance penalty of pretty printing the response content in the exception case. Otherwise I would leave it as-is.